GUIDELINES.md

```diff
@@ -0,0 +1,79 @@
+# 📑 Engineering Guidelines
+
+## ✅ Testing
+
+**Don't optimise for coverage, optimise for well-designed tests.**
+
+Write positive and negative unit tests.
+
+- Write positive unit tests for things that the code should handle. Validate any states that change as a result of these tests.
+- Write negative unit tests for things that the code should not handle. It is helpful to follow up on the positive test (as an adjacent test) and make the change needed to make it pass.
+- Each code path should have its own unit test.
+
+Any addition or change to the code must be accompanied by relevant and comprehensive tests. Refactors should avoid simultaneous changes to tests.
+
+The test suite should run automatically for each change in the repository, and for pull requests, the tests must succeed before merging.
+
+Please consider writing [Foundry](https://github.com/foundry-rs/foundry)-based unit tests, property-based tests (i.e. fuzzing), and invariant tests for all contracts, if applicable.
+
+## 🪅 Code Style
+
+🐍Vyper code should be written in a consistent format that follows our [🐍Vyper Conventions](#vyper-conventions).
+
+Solidity test code should be written in a consistent format enforced by a linter and prettier that follows the official [Solidity Style Guide](https://docs.soliditylang.org/en/latest/style-guide.html). Also, we refer to Foundry's [best practices](https://book.getfoundry.sh/tutorials/best-practices?highlight=security#best-practices).
+
+The code should be simple and straightforward, with a focus on readability and comprehensibility. Consistency and predictability should be maintained throughout the code base. This is especially true for naming, which should be systematic, clear, and concise.
+
+Wherever possible, modularity, composability, and gas efficiency should be pursued, but not at the expense of security compromises.
+
+## 🛠 Pull Request (PR) Format
+
+Pull requests are _squash-merged_ in most cases to keep the `main` branch history clean. The title of the PR becomes the commit message, so it should be written in a consistent format:
+
+- Start with an emoji that well describes the PR.
+- Use capitalisation for the PR title: "✨ Add Feature X" and not "✨ Add feature x".
+- Do not end with a full stop (i.e. period).
+- Write in the imperative: "✨ Add Feature X" and not "✨ Adds Feature X" or "✨ Added Feature X".
+
+This repository does not follow conventional commits, so do not prefix the title with "fix:", "feat:", or similar. Also, pull requests in progress should be submitted as _Drafts_ and should not be prefixed with "WIP:" or similar.
+
+Branch names do not matter, and commit messages within a PR are mostly not important either, although they can support the review process.
+
+## 🐍Vyper Conventions
+
+- The names of `constant`, `immutable`, and state variables, functions, and function parameters use the snake case notation (e.g. `my_function`) if no other notation is enforced via an EIP standard.
```

Comment:
constants and immutable values should be in "screaming snake case"

Reply:
noted and enhanced